viewer/src/datamodels/cad/sector/CachedRepository.ts

```diff
@@ -107,24 +116,79 @@ export class CachedRepository implements Repository {
   // unless we want them to eventually set their priority to lower in the cache.
 
   loadSector(): OperatorFunction<WantedSector, ConsumedSector> {
-    return pipe(
-      this._loadingCounter.incrementOnNext(),
-      flatMap(sector => {
-        if (this._consumedSectorCache.has(this.wantedSectorCacheKey(sector))) {
-          return this._consumedSectorCache.get(this.wantedSectorCacheKey(sector));
-        } else if (sector.levelOfDetail == LevelOfDetail.Detailed) {
-          return this.loadDetailedSectorFromNetwork(sector);
-        } else if (sector.levelOfDetail == LevelOfDetail.Simple) {
-          return this.loadSimpleSectorFromNetwork(sector);
-        } else if (sector.levelOfDetail == LevelOfDetail.Discarded) {
-          return of(sector).pipe(map(wantedSector => ({ ...wantedSector, group: undefined } as ConsumedSector)));
-        } else {
-          throw new Error('Unhandled case');
-        }
-      }, this._concurrentNetworkOperations),
-      this._loadingCounter.decrementOnNext(),
-      this._loadingCounter.resetOnComplete()
-    );
+    return (source$: Observable<WantedSector>) => {
+      /* Split wantedSectors into a pipe of discarded wantedSectors and a pipe of simple and detailed wantedSectors.
+       * ----------- wantedSectors -----------------------
+       * \---------- discarded wantedSectors -------------
+       *  \--------- simple and detailed wantedSectors ---
+       */
```

Comment:
Love this type of comments!

Reply:
I thought it was about time. The uncertainty of rx within the team really makes me want to do a little presentation, maybe a tech talk later if you feel the presentation is worth bringing to others.